datastream-server-restli/src/main/java/com/linkedin/datastream/server/dms/DatastreamResources.java

```diff
@@ -739,6 +740,51 @@ public ActionResult<Void> resumeSourcePartitions(@PathKeysParam PathKeys pathKey
     return new ActionResult<>(HttpStatus.S_200_OK);
   }
 
+  /**
+   * This endpoint handles the management of throughput violating topics. These topics' metrics and
+   * SLAs are reported separately as their throughput are not within brooklin's permissible bounds.
+   * @param pathKeys Datastream resource key
+   * @param throughputViolations StringArray of list of topics
+   * */
+  @Action(name = "reportThroughputViolatingTopics", resourceLevel = ResourceLevel.ENTITY)
+  public ActionResult<Void> reportThroughputViolatingTopics(@PathKeysParam PathKeys pathKeys,
+      @ActionParam("throughputViolations") StringArray throughputViolations) {
+
+    // Get datastream.
+    String datastreamName = pathKeys.getAsString(KEY_NAME);
+    // Log for debugging purposes.
+    LOG.info("reportThroughputViolatingTopics called for datastream: {}, with topics: {}", datastreamName,
+        throughputViolations);
+
+    // Null check
+    if (throughputViolations == null) {
+      LOG.error("NULL input for throughput violating topics not supported.");
+      return new ActionResult<>(HttpStatus.S_400_BAD_REQUEST);
+    }
```

Comment:
Validate if the datastream exists or not. 
Add a test to pass input as incorrect datastream.